ratik/src/main/java/org/opencadc/ratik/InventoryValidator.java

```diff
@@ -150,7 +169,215 @@ public InventoryValidator(Map<String, Object> daoConfig, URI resourceID, Artifac
+                    if (remoteArtifact == null) {
+                        remoteArtifact = remoteIterator.hasNext() ? remoteIterator.next() : null;
+                    }
+                    // TODO sanity check? if either iterator has no results in the first loop, exit?
+                    if (localArtifact == null && remoteArtifact == null) {
+                        artifactsToValidate = false;
+                        continue;
+                    }
+                    log.debug(String.format("comparing Artifacts:\n local - %s\nremote - %s",
+                                            localArtifact, remoteArtifact));
+
+                    // check if Artifacts are the same, or if the local Artifact
+                    // precedes or follows the remote Artifact.
+                    int order = orderArtifacts(localArtifact, remoteArtifact);
+                    if (order == 0) {
+                        validate(localArtifact, remoteArtifact);
+                        localArtifact = null;
+                        remoteArtifact = null;
+                    } else if (order < 0) {
+                        validate(localArtifact, null);
+                        localArtifact = null;
+                    } else {
+                        validate(null, remoteArtifact);
+                        remoteArtifact = null;
+                    }
+                }
+            } catch (IOException e) {
+                log.error("Error closing iterator: " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Useful for overriding in tests.
+     *
+     * @param localArtifact the local Artifact.
+     * @param remoteArtifact the remote Artifact.
+     */
+    void validate(Artifact localArtifact, Artifact remoteArtifact) {
+        log.debug(String.format("validating:\n local - %s\nremote - %s", localArtifact, remoteArtifact));
+        artifactValidator.validate(localArtifact, remoteArtifact);
+    }
+
+    /**
+     * Order two Artifacts on the String representation of Artifact.uri.
+     * Must match the ordering of a postgresql ORDER BY ASC on Artifact.uri.
+     * - if local equals remote returns 0.
+     * - if local lexicographically precedes remote returns a negative value.
+     * - if local lexicographically follows remote returns a positive value.
+     */
+    int orderArtifacts(Artifact localArtifact, Artifact remoteArtifact) {
+        log.debug(String.format("order artifact uri's:\n local - %s\nremote - %s",
+                                localArtifact == null ? "null" : localArtifact.getURI(),
+                                remoteArtifact == null ? "null" : remoteArtifact.getURI()));
+        int result;
+        if (localArtifact == null) {
+            result = 1;
+        } else if (remoteArtifact == null) {
+            result = -1;
+        } else {
+            result = localArtifact.getURI().toString().compareTo(remoteArtifact.getURI().toString());
+        }
+        if (log.isDebugEnabled()) {
```

Comment:
This debugging isn't useful - the unit test of this method would be. debug messages in there would be fine if they help

Reply:
Removed the debug message. Doesn't help for unit tests.